### PR TITLE
Reader document list formatting and some minor refactors

### PR DESCRIPTION
**Pull request by tejans24**
connects (#1527)
connects (#1528)

### CODE REVIEW

**Comment by tejans24** [comment]
Keeping font awesome icons until we have svgs

File: client/app/reader/PdfListView.jsx [hunk 1]
```diff
@@ -17,7 +15,12 @@ export default class PdfListView extends React.Component {
       className = "fa-caret-up";
     }
 
-    let sortIcon = <i className={`fa ${className}`} aria-hidden="true"></i>;
+    let sortIcon = <i className={`fa fa-1 ${className} table-icon`}
+      aria-hidden="true"></i>;
+    let filterIcon = <i className="fa fa-1 fa-filter table-icon bordered-icon"
+      aria-hidden="true"></i>;
+    let notsortedIcon = <i className="fa fa-1 fa-arrows-v table-icon"
```
Comment (tejans24) [inline client/app/reader/PdfListView.jsx]: adding this as a placeholder for knowing which rows can be sorted. This icon currently isn't a great look.

File: client/app/components/SearchBar.jsx [hunk 2]
```diff
@@ -16,6 +16,7 @@ export default class SearchBar extends React.Component {
       <label className="usa-sr-only" htmlFor={id}>Search small</label>
       <input id={id} onChange={this.onChange} type="search" name="search" value={value}/>
       <button onClick={onClick} type="submit">
+        <i className="fa fa-search" aria-hidden="true"></i>
```
Comment (mdbenjam) [inline client/app/components/SearchBar.jsx]: So backstory: This searchbar should already have that symbol. It disappeared about two months ago when this style was introduced to caseflow-commons: 

```
.usa-search [type="submit"], .usa-search .usa-search-submit {
    background-image: none !important;
}
```

I have no idea why this was introduced, but it should not be there... I'm making a change to commons already, so I can just add this to it. And we should be able to remove this icon.

File: client/app/components/reader/DocumentListHeader.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import React, { PropTypes } from 'react';
+import SearchBar from '../SearchBar';
+import Button from '../Button';
+
+const DocumentListHeader = (props) => {
```
Comment (mdbenjam) [inline client/app/components/reader/DocumentListHeader.jsx]: I like breaking this out into its own component 👍

File: client/app/reader/PdfListView.jsx [hunk 4]
```diff
@@ -32,51 +35,81 @@ export default class PdfListView extends React.Component {
     // is triggered when a middle mouse button is clicked while onClick isn't.
     return [
       {
+        header: <div
+          id="categories-header"
+          className="document-list-header-categories"
```
Comment (mdbenjam) [inline client/app/reader/PdfListView.jsx]: Is this class defined anywhere?

Comment (tejans24) [inline client/app/reader/PdfListView.jsx]: Not yet, this is related to #1538

File: client/app/components/SearchBar.jsx (hunk 2 above)
Comment (tejans24) [inline client/app/components/SearchBar.jsx]: kk, I will remove this icon then

File: app/assets/stylesheets/reader/_document_list.scss [hunk 5]
```diff
@@ -0,0 +1,52 @@
+$bordered-icon-border-color: #808080;
```
Comment (tejans24) [inline app/assets/stylesheets/reader/_document_list.scss]: I made the change to use $color-gray

File: app/assets/stylesheets/reader/_document_list.scss [hunk 6]
```diff
@@ -0,0 +1,54 @@
+@import 'uswds/lib/addons/font-stacks';
+@import 'uswds/core/variables';
+
+$gray-font-color: #6b7079;
+$tag-backgrond-color: #d6d7d9;
```
Comment (mdbenjam) [inline app/assets/stylesheets/reader/_document_list.scss]: Ideally I'd still like to get rid of these values if there is another color the design team could use. But it's not super important for now. Maybe make a tech improvement ticket to see if we can remove them later.

File: client/app/reader/PdfListView.jsx (hunk 4 above)
Comment (mdbenjam) [inline client/app/reader/PdfListView.jsx]: 👍

**Pull request merged**